apps/ybal/contexts/useYBal.tsx

```diff
@@ -0,0 +1,235 @@
+			veEscrowContract.balanceOf(VECRV_YEARN_TREASURY_ADDRESS),
+			veEscrowContract.totalSupply(),
+			yBalContract.totalSupply(),
+			styBalContract.totalAssets(),
+			lpyBalContract.totalSupply(),
+			crvYCRVLpContract.get_dy(1, 0, ethers.constants.WeiPerEther)
+		]) as [BigNumber, BigNumber, BigNumber, BigNumber, BigNumber, BigNumber, BigNumber, BigNumber];
+
+		return ({
+			['legacy']: yveBalTotalSupply.sub(yveCRVInYCRV),
+			['treasury']: veBalBalance.sub(yveBalTotalSupply.sub(yveCRVInYCRV)).sub(yCRVTotalSupply),
+			['yBalSupply']: yCRVTotalSupply,
+			['styBalSupply']: styCRVTotalSupply,
+			['lpyBalSupply']: lpyCRVTotalSupply,
+			['balYBalPeg']: balYBalPeg,
+			['boostMultiplier']: veBalBalance.mul(1e4).div(styCRVTotalSupply),
+			['veBalTotalSupply']: veBalTotalSupply,
+			['veBalBalance']: veBalBalance
+		});
+	}, [provider]);
+	const	{data: holdings} = useSWR('numbers', numbersFetchers, {shouldRetryOnError: false});
+
+
+	/* 🔵 - Yearn Finance ******************************************************
+	**	Once the wallet is connected and a provider is available, we can fetch
+	**	the allowance informations for a specific wallet. As the possible path
+	**	are limited, we can hardcode the contract addresses.
+	***************************************************************************/
+	const getAllowances = useCallback(async (): Promise<TDict<BigNumber>> => {
+		if (!isActive || !provider) {
+			return {};
+		}
+		const	currentProvider = provider || getProvider(1);
+		const	ethcallProvider = await newEthCallProvider(currentProvider);
+		const	userAddress = address;
+		const	yBalContract = new Contract(YBAL_TOKEN_ADDRESS as string, YVECRV_ABI);
+		const	styBalContract = new Contract(STYBAL_TOKEN_ADDRESS as string, YVECRV_ABI);
+		const	lpyBalContract = new Contract(LPYBAL_TOKEN_ADDRESS as string, YVECRV_ABI);
+		const	yveCRVContract = new Contract(YVECRV_TOKEN_ADDRESS as string, YVECRV_ABI);
+		const	balContract = new Contract(BAL_TOKEN_ADDRESS as string, ERC20_ABI);
+		const	yvBoostContract = new Contract(YVBOOST_TOKEN_ADDRESS as string, ERC20_ABI);
+		const	yBalPoolContract = new Contract(YBAL_BALANCER_POOL_ADDRESS as string, YVECRV_ABI);
+
+		const	[
+			yCRVAllowanceZap, styCRVAllowanceZap, lpyCRVAllowanceZap,
+			yveCRVAllowanceZap, crvAllowanceZap, yvBoostAllowanceZap,
+			yveCRVAllowanceLP, crvAllowanceLP,
+			yCRVPoolAllowanceVault
+		] = await ethcallProvider.tryAll([
+			yBalContract.allowance(userAddress, ZAP_YEARN_VE_CRV_ADDRESS),
+			styBalContract.allowance(userAddress, ZAP_YEARN_VE_CRV_ADDRESS),
+			lpyBalContract.allowance(userAddress, ZAP_YEARN_VE_CRV_ADDRESS),
+			yveCRVContract.allowance(userAddress, ZAP_YEARN_VE_CRV_ADDRESS),
+			balContract.allowance(userAddress, ZAP_YEARN_VE_CRV_ADDRESS),
+			yvBoostContract.allowance(userAddress, ZAP_YEARN_VE_CRV_ADDRESS),
+			yveCRVContract.allowance(userAddress, YVECRV_POOL_LP_ADDRESS),
+			balContract.allowance(userAddress, YVECRV_POOL_LP_ADDRESS),
```

Comment:
Is this one of the things missing? I think it would be good to add a few TODOs on these lines so we have an easier time adding what's missing